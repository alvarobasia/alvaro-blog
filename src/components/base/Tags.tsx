import React from 'react';

interface TagPros {
  tag: string;
}

const Tags: React.FC<TagPros> = ({tag}: TagPros) => {
  return (
    <div
      key={tag}
      className="px-2 border-2 rounded-2xl border-blue-500 cursor-pointer group hover:border-white"
    >
      <span className="text-blue-500 text-sm group-hover:text-white transition-all">
        #{tag}
      </span>
    </div>
  );
};

export default Tags;
