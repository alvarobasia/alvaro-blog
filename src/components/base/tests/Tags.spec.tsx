import React from 'react';

import {render, screen} from '@testing-library/react';

import Tags from '../Tags';

describe('Test Tags component', () => {
  it('should render Avatar component', () => {
    const {baseElement} = render(<Tags tag="teste" />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should show correct tag', () => {
    render(<Tags tag="teste" />);
    const text = screen.getByText('#teste');
    expect(text).toBeInTheDocument();
  });
});
