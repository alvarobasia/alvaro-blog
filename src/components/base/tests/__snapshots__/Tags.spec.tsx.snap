// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Tags component > should render Avatar component 1`] = `
<body>
  <div>
    <div
      class="px-2 border-2 rounded-2xl border-blue-500 cursor-pointer group hover:border-white"
    >
      <span
        class="text-blue-500 text-sm group-hover:text-white transition-all"
      >
        #
        teste
      </span>
    </div>
  </div>
</body>
`;
